### PR TITLE
Edit and move doc for AOT compiler options

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Compiler options are documented in the AOT compiler page.

Issue Number: N/A

## What is the new behavior?
Move the options doc to a separate page (and edit content).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/angular-compiler-options.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      "angularCompilerOptions": {
+        "fullTemplateTypeCheck": true,
+        "preserveWhitespaces": true,
+                  ...
+      }
+  }
+  ```
+This page describes the available Angular template compiler options.
+
+### `allowEmptyCodegenFiles`
```
Comment (alan-agius4) [inline aio/content/guide/angular-compiler-options.md]: I don’t think that titles should be in backticks.

File: aio/content/guide/angular-compiler-options.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+# Angular compiler options
+
```
Comment (alan-agius4) [inline aio/content/guide/angular-compiler-options.md]: Isn’t this section a duplicate of https://pr31829-2ea6b2a.ngbuilds.io/guide/aot-compiler#controlling-app-compilation?

Comment (alan-agius4) [reply aio/content/guide/angular-compiler-options.md]: I think we should remove the duplicate sections from the latter.

Comment (jbogarthyde) [reply aio/content/guide/angular-compiler-options.md]: oops - that was the whole point!

File: aio/content/guide/angular-compiler-options.md (hunk 1 above)
Comment (jbogarthyde) [reply aio/content/guide/angular-compiler-options.md]: Normally that's the case, but this is really a reference page for the options. It should actually be in table format, but some of the descriptions are too long. By making them headings, there is a right-hand index to the code options. I think it's a valid exception to the general rule for titles.

**Review by alan-agius4 — APPROVED** [review]

**Comment by jbogarthyde** [comment]
Just needs code-owner approval.

**Pull request closed without merging**